Load cmhc data

OussemaTrabelsi98 opened the pull request:
This PR introduces a function to fetch and clean data from the CMHC API at the CMA level, supporting multiple series and dimensions dynamically. It also structures metadata as a dictionary.

bdbmax commented:
Thanks Oussama, great PR. I just have one thing in mind and it's about the naming conventions of functions + the exporting of them. If they are related to the CMHC extract, let's name them all with a start with `cmhc_` similar to how we do for `census_` functions. And when a function is only meant to be used within the package rather than externally, like a utility function you use to process results, split cma, clean results, etc., you don't have to use the @export tag in the roxygen2 documentation. those functions will still be usable by doing cc.data:::thefunction, but not using cc.data::thefunction. They won't clutter the namespace!

OussemaTrabelsi98 commented:
Thank you, Max, for the comment. I’ll update this right now!

Le lun. 24 mars 2025 à 15:06, Maxime Bélanger De Blois <
***@***.***> a écrit :

> Thanks Oussama, great PR. I just have one thing in mind and it's about the
> naming conventions of functions + the exporting of them. If they are
> related to the CMHC extract, let's name them all with a start with cmhc_
> similar to how we do for census_ functions. And when a function is only
> meant to be used within the package rather than externally, like a utility
> function you use to process results, split cma, clean results, etc., you
> don't have to use the @export <https://github.com/export> tag in the
> roxygen2 documentation. those functions will still be usable by doing
> cc.data:::thefunction, but not using cc.data::thefunction. They won't
> clutter the namespace!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Curbcut/cc.data/pull/1#issuecomment-2748249487>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BQBUZ5DJMDEQBGR5LKRLN7L2WAGM3AVCNFSM6AAAAABZHW3CNGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBYGI2DSNBYG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: bdbmax]*bdbmax* left a comment (Curbcut/cc.data#1)
> <https://github.com/Curbcut/cc.data/pull/1#issuecomment-2748249487>
>
> Thanks Oussama, great PR. I just have one thing in mind and it's about the
> naming conventions of functions + the exporting of them. If they are
> related to the CMHC extract, let's name them all with a start with cmhc_
> similar to how we do for census_ functions. And when a function is only
> meant to be used within the package rather than externally, like a utility
> function you use to process results, split cma, clean results, etc., you
> don't have to use the @export <https://github.com/export> tag in the
> roxygen2 documentation. those functions will still be usable by doing
> cc.data:::thefunction, but not using cc.data::thefunction. They won't
> clutter the namespace!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Curbcut/cc.data/pull/1#issuecomment-2748249487>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BQBUZ5DJMDEQBGR5LKRLN7L2WAGM3AVCNFSM6AAAAABZHW3CNGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBYGI2DSNBYG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
